Index: u-boot-2022.10/cmd/ubi.c
===================================================================
--- u-boot-2022.10.orig/cmd/ubi.c
+++ u-boot-2022.10/cmd/ubi.c
@@ -110,15 +110,21 @@ static int verify_mkvol_req(const struct
 	int n, err = EINVAL;
 
 	if (req->bytes < 0 || req->alignment < 0 || req->vol_type < 0 ||
-	    req->name_len < 0)
+	    req->name_len < 0) {
+		printf("%d check failed!\n", __LINE__);
 		goto bad;
+	}
 
-	if ((req->vol_id < 0 || req->vol_id >= ubi->vtbl_slots) &&
-	    req->vol_id != UBI_VOL_NUM_AUTO)
+	if ((req->vol_id < 0) &&
+	    req->vol_id != UBI_VOL_NUM_AUTO) {
+		printf("Invalid UBI id %d!\n", req->vol_id);
 		goto bad;
+	}
 
-	if (req->alignment == 0)
+	if (req->alignment == 0) {
+		printf("Invalid alignment of 0!\n");
 		goto bad;
+	}
 
 	if (req->bytes == 0) {
 		printf("No space left in UBI device!\n");
@@ -127,15 +133,21 @@ static int verify_mkvol_req(const struct
 	}
 
 	if (req->vol_type != UBI_DYNAMIC_VOLUME &&
-	    req->vol_type != UBI_STATIC_VOLUME)
+	    req->vol_type != UBI_STATIC_VOLUME) {
+		printf("Invalid ubi type %d!\n", req->vol_type);
 		goto bad;
+	}
 
-	if (req->alignment > ubi->leb_size)
+	if (req->alignment > ubi->leb_size) {
+		printf("alignment %d > ubi leb_size %d!\n", req->alignment, ubi->leb_size);
 		goto bad;
+	}
 
 	n = req->alignment % ubi->min_io_size;
-	if (req->alignment != 1 && n)
+	if (req->alignment != 1 && n) {
+		printf("alignment %d %% %d min_io_size check failed!\n", req->alignment, ubi->min_io_size);
 		goto bad;
+	}
 
 	if (req->name_len > UBI_VOL_NAME_MAX) {
 		printf("Name too long!\n");
