Index: u-boot-2022.10/cmd/ubi.c
===================================================================
--- u-boot-2022.10.orig/cmd/ubi.c
+++ u-boot-2022.10/cmd/ubi.c
@@ -89,6 +89,21 @@ static int ubi_check_volumename(const st
 	return strcmp(vol->name, name);
 }
 
+static int ubi_checkid(int vol_id)
+{
+	int i;
+
+	for (i = 0; i < (ubi->vtbl_slots + 1); i++) {
+		if (!ubi->volumes[i])
+			continue;	/* Empty record */
+
+		if (ubi->volumes[i]->vol_id == vol_id)
+			return 0;
+	}
+
+	return 1;
+}
+
 static int ubi_check(char *name)
 {
 	int i;
@@ -198,6 +213,21 @@ int ubi_create_vol(char *volume, int64_t
 	return ubi_create_volume(ubi, &req);
 }
 
+struct ubi_volume *ubi_find_volume_id(int vol_id)
+{
+	struct ubi_volume *vol = NULL;
+	int i;
+
+	for (i = 0; i < ubi->vtbl_slots; i++) {
+		vol = ubi->volumes[i];
+		if (vol && vol->vol_id == vol_id)
+			return vol;
+	}
+
+	printf("Volume with id %d not found!\n", vol_id);
+	return NULL;
+}
+
 struct ubi_volume *ubi_find_volume(char *volume)
 {
 	struct ubi_volume *vol = NULL;
@@ -305,6 +335,48 @@ static int ubi_rename_vol(char *oldname,
 	return ubi_rename_volumes(ubi, &list);
 }
 
+static int ubi_setname_vol(int vol_id, char *newname)
+{
+	struct ubi_volume *vol;
+	struct ubi_rename_entry rename;
+	struct ubi_volume_desc desc;
+	struct list_head list;
+
+	vol = ubi_find_volume_id(vol_id);
+	if (!vol) {
+		printf("%s: volume with id %d doesn't exist\n", __func__, vol_id);
+		return ENODEV;
+	}
+
+	/* already has the same */
+	if (!ubi_check_volumename(vol, newname))
+		return 0;
+
+	if (!ubi_check(newname)) {
+		printf("%s: volume %s already exist\n", __func__, newname);
+		return EINVAL;
+	}
+
+	printf("Rename UBI volume %s (%d) to %s\n", vol->name, vol_id, newname);
+
+	if (ubi->ro_mode) {
+		printf("%s: ubi device is in read-only mode\n", __func__);
+		return EROFS;
+	}
+
+	rename.new_name_len = strlen(newname);
+	strcpy(rename.new_name, newname);
+	rename.remove = 0;
+	desc.vol = vol;
+	desc.mode = 0;
+	rename.desc = &desc;
+	INIT_LIST_HEAD(&rename.list);
+	INIT_LIST_HEAD(&list);
+	list_add(&rename.list, &list);
+
+	return ubi_rename_volumes(ubi, &list);
+}
+
 static int ubi_volume_continue_write(char *volume, void *buf, size_t size)
 {
 	int err = 1;
@@ -602,6 +674,35 @@ static int do_ubi(struct cmd_tbl *cmdtp,
 		return ubi_info(layout);
 	}
 
+	if (strcmp(argv[1], "checkid") == 0) {
+        int vol_id;
+		if (argc < 2) {
+			printf("Error, no volume id passed\n");
+			return 1;
+		}
+
+		vol_id = simple_strtoull(argv[2], NULL, 16);
+		return ubi_checkid(vol_id);
+	}
+
+	if (strcmp(argv[1], "name") == 0) {
+		int vol_id;
+		struct ubi_volume *vol;
+		if (argc < 2) {
+			printf("Error, no volume id passed\n");
+			return 1;
+		}
+
+		vol_id = simple_strtoull(argv[2], NULL, 16);
+		vol = ubi_find_volume_id(vol_id);
+		if (!vol) {
+			printf("Error, couldn't find the volume with id %d", vol_id);
+			return 1;
+		}
+		printf(vol->name);
+		return 0;
+	}
+
 	if (strcmp(argv[1], "check") == 0) {
 		if (argc > 2)
 			return ubi_check(argv[2]);
@@ -666,6 +767,11 @@ static int do_ubi(struct cmd_tbl *cmdtp,
 	if (IS_ENABLED(CONFIG_CMD_UBI_RENAME) && !strncmp(argv[1], "rename", 6))
 		return ubi_rename_vol(argv[2], argv[3]);
 
+	if (IS_ENABLED(CONFIG_CMD_UBI_RENAME) && !strncmp(argv[1], "setname", 6)) {
+		int vol_id = simple_strtoull(argv[2], NULL, 16);
+		return ubi_setname_vol(vol_id, argv[3]);
+    }
+
 	if (strncmp(argv[1], "skipcheck", 9) == 0) {
 		/* E.g., change skip_check flag */
 		if (argc == 4) {
@@ -743,6 +849,10 @@ U_BOOT_CMD(
 		" - Display volume and ubi layout information\n"
 	"ubi check volumename"
 		" - check if volumename exists\n"
+	"ubi checkid volumeid"
+		" - check if volume with id exists\n"
+	"ubi name volumeid"
+		" - prints the name of the volume id\n"
 	"ubi create[vol] volume [size] [type] [id] [--skipcheck]\n"
 		" - create volume name with size ('-' for maximum"
 		" available size)\n"
@@ -756,6 +866,8 @@ U_BOOT_CMD(
 		" - Remove volume\n"
 #if IS_ENABLED(CONFIG_CMD_UBI_RENAME)
 	"ubi rename oldname newname\n"
+	"ubi setname volumeid newname"
+		" - set the name of the volume identified by volumeid\n"
 #endif
 	"ubi skipcheck volume on/off - Set or clear skip_check flag in volume header\n"
 	"[Legends]\n"
